src/ai/susi/SusiServer.java

```diff
@@ -626,33 +626,27 @@ private static void checkServerPorts(int httpPort, int httpsPort) throws IOExcep
     	// check http port
         if(!httpsMode.equals(HttpsMode.ONLY)){
 	        ServerSocket ss = null;
-	        try {
-	            ss = new ServerSocket(httpPort);
-	            ss.setReuseAddress(true);
-	            ss.setReceiveBufferSize(65536);
-	        } catch (IOException e) {
-	            // the socket is already occupied by another service
-	            throw new IOException("port " + httpPort + " is already occupied by another service, maybe another SUSI is running on this port already. exit.");
-	        } finally {
-	            // close the socket again
-	            if (ss != null) ss.close();
-	        }
+            checkPort(httpPort, ss);
         }
-        
+
         // check https port
         if(httpsMode.isGreaterOrEqualTo(HttpsMode.ON)){
 	        ServerSocket sss = null;
-	        try {
-	            sss = new ServerSocket(httpsPort);
-	            sss.setReuseAddress(true);
-	            sss.setReceiveBufferSize(65536);
-	        } catch (IOException e) {
-	            // the socket is already occupied by another service
-	        	throw new IOException("port " + httpsPort + " is already occupied by another service, maybe another SUSI is running on this port already. exit.");
-	        } finally {
-	            // close the socket again
-	            if (sss != null) sss.close();
-	        }
+            checkPort(httpsPort, sss);
```

Comment:
Same as https://github.com/fossasia/susi_server/pull/561/files?diff=split#diff-d93e29e1c875d30ce3d763c19cce4084R629

Reply:
Can you explain more? @yukiisbored